### PR TITLE
Rewrote Dagon and Added Spell Amp

### PR DESCRIPTION
**Pull request by Chronophylos**
Added more Stats to Modifier Generic Bonus

### CODE REVIEW

File: game/scripts/vscripts/items/dagon.lua [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  local particleName = "particles/items/dagon_oaa.vpcf"
+  local particleThickness = 300 + (100 * level) --Control Point 2 in Dagon's particle effect takes a number between 400 and 2000, depending on its level.
+
+  local damage = self:GetSpecialValueFor("damage")
+
+  local particle = ParticleManager:CreateParticle(particleName,  PATTACH_POINT_FOLLOW, caster)
+  ParticleManager:SetParticleControlEnt(particle, 0, caster, PATTACH_POINT_FOLLOW, "attach_attack1", caster:GetOrigin(), true)
+  ParticleManager:SetParticleControlEnt(particle, 1, target, PATTACH_POINT_FOLLOW, "attach_hitloc", target:GetOrigin(), true)
+  ParticleManager:SetParticleControl(particle, 2, Vector(particleThickness))
```
Comment (Trildar) [inline game/scripts/vscripts/items/dagon.lua]: I don't know if it's actually required, but I've always done `ParticleManager:ReleaseParticleIndex` after I'm done with a particle.

Comment (Trildar) [inline game/scripts/vscripts/items/dagon.lua]: There also seems to be some extra particle that Dagon 5 applies to its target or something, but I'm not sure how to use it.

File: game/scripts/vscripts/items/dagon.lua [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+
+  caster:EmitSound("DOTA_Item.Dagon.Activate")
+  target:EmitSound("DOTA_Item.Dagon5.Target")
```
Comment (Trildar) [inline game/scripts/vscripts/items/dagon.lua]: Only doing the `Dagon5` sound for level 5 and up may be more accurate, but it's not really that important.

Comment (Chronophylos) [inline game/scripts/vscripts/items/dagon.lua]: In Pizzalol's Spell Library they were using both

File: game/scripts/vscripts/items/dagon.lua (hunk 1 above)
Comment (Chronophylos) [inline game/scripts/vscripts/items/dagon.lua]: I've never used it
but it may be better to use

File: game/scripts/vscripts/items/dagon.lua (hunk 2 above)
Comment (Trildar) [inline game/scripts/vscripts/items/dagon.lua]: Well, the Bristleback implementation wasn't entirely accurate either with the damage storage, though to be fair, that one had a comment explicitly saying the author wasn't sure if that's how Bristleback worked.

Comment (Chronophylos) [inline game/scripts/vscripts/items/dagon.lua]: Maybe `DOTA_Item.Dagon.Activate` is always played and `DOTA_Item.Dagon5.Target` for Level 5 and above as a additional sound

Comment (Chronophylos) [inline game/scripts/vscripts/items/dagon.lua]: Cast Range is increasing too and both sounds have a limited distance

**Pull request merged**